### PR TITLE
chain: MsgCreateDataSource &  MsgEditDataSource

### LINKED ISSUE #453: Create MsgCreateDataSource and MsgEditDataSource struct
See [here](https://github.com/bandprotocol/d3n/tree/master/spec#msgcreatedatasource) for message specification. For this issue, please define the message struct with proper tests. The handler of the message can just log some message and will be implemented later.

### PR DESCRIPTION
**Pull request by evilpeach**
Fixes #453 

- [x]  MsgCreateDataSource
- [x] Test case for MsgCreateDataSource
- [x] MsgEditDataSource
- [x] Test case for MsgEditDataSource

### CODE REVIEW

File: chain/x/zoracle/internal/types/msgs.go [hunk 1]
```diff
@@ -221,3 +221,144 @@ func (msg MsgDeleteCode) GetSignBytes() []byte {
...
+		Executable: executable,
+		Sender:     sender,
+	}
+}
+
+// Route should return the name of the module
+func (msg MsgCreateDataSource) Route() string { return RouterKey }
+
+// Type should return the action
+func (msg MsgCreateDataSource) Type() string { return "createDataSource" }
```
Comment (taobun) [inline chain/x/zoracle/internal/types/msgs.go]: Use snake case for type

File: chain/x/zoracle/internal/types/msgs.go [hunk 2]
```diff
@@ -221,3 +221,144 @@ func (msg MsgDeleteCode) GetSignBytes() []byte {
...
+		Executable:   executable,
+		Sender:       sender,
+	}
+}
+
+// Route should return the name of the module
+func (msg MsgEditDataSource) Route() string { return RouterKey }
+
+// Type should return the action
+func (msg MsgEditDataSource) Type() string { return "editDataSource" }
```
Comment (taobun) [inline chain/x/zoracle/internal/types/msgs.go]: ditto

**Review by taobun — CHANGES_REQUESTED** [review]

File: chain/x/zoracle/internal/types/msgs.go [hunk 3]
```diff
@@ -221,3 +221,138 @@ func (msg MsgDeleteCode) GetSignBytes() []byte {
...
+	return []sdk.AccAddress{msg.Sender}
+}
+
+// GetSignBytes implements the sdk.Msg interface for MsgCreateDataSource.
+func (msg MsgCreateDataSource) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(msg)
+	return sdk.MustSortJSON(bz)
+}
+
+// MsgEditDataSource is a message for editing existing data source.
```
Comment (sorawit) [inline chain/x/zoracle/internal/types/msgs.go]: nit: editing an existing

**Review by sorawit — CHANGES_REQUESTED** [review]
Small grammar change

**Review by taobun — APPROVED** [review]

**Review by sorawit — APPROVED** [review]

**Pull request merged**